components/net/cookie_storage.rs

```diff
@@ -65,7 +78,61 @@ impl CookieStorage {
         }
 
         // Step 12
-        self.cookies.push(cookie);
+        let domain = reg_host(&cookie.cookie.domain.clone().unwrap_or("".to_owned())).unwrap_or("".to_owned());
+        if !self.cookies_map.contains_key(&domain) {
+        let tempdomain = domain.clone();
+            let cookies: Vec<Cookie> = Vec::new();
+            self.cookies_map.insert(tempdomain, cookies);
+        }
+        let mut cookies = self.cookies_map.get_mut(&domain).unwrap();
+
+        if cookies.len() == self.max_per_host {
```

Comment:
Could we move all of the eviction implementation into a `evict_one_cookie` helper method?